docs(native): Add documentation for REST-based remote functions (#26513)
prestodb/presto: presto-docs/src/main/sphinx/presto_cpp/features.rst

``remote-function-server.thrift.address`` and
``remote-function-server.thrift.port``.

REST-Based Remote Functions

> ----------------------------

Reviewer: ```suggestion
---------------------------
```

In addition to Thrift-based remote function execution, Presto C++ also supports
REST-based remote functions that communicate via HTTP/REST protocol. This provides
an alternative approach to remote function execution that may be easier to
implement and integrate with existing REST-based services.

REST functions use HTTP POST requests to invoke remote functions and receive
results. The communication uses serialized data in either Presto page format
or Spark unsafe row format for efficient data transfer.

Configuration
^^^^^^^^^^^^^

To enable REST-based remote functions, configure the following property:

``remote-function-server.rest.url``

> """"""""""""""""""""""""""""""""""""

Reviewer: ```suggestion
"""""""""""""""""""""""""""""""""""
```

* **Type:** ``string``
* **Default value:** ``""``

The base URL of the REST server that hosts remote functions. When specified,
the Presto C++ worker will invoke functions via HTTP REST endpoints at this

> server. The URL should include the protocol and host (e.g.,

Reviewer: ```suggestion
server. The URL should include the protocol and host (for example,
```
Avoid Latin and Latin abbreviations. See the [Gitlab documentation style guide recommended word list entry for e.g ](https://docs.gitlab.com/development/documentation/styleguide/word_list/#eg)for discussion and alternative phrasings.

Supported values:

* ``presto_page``: Uses Presto's native page serialization format
* ``spark_unsafe_row``: Uses Spark's unsafe row serialization format

Setup and Usage
^^^^^^^^^^^^^^^

To use REST-based remote functions in your Presto C++ cluster:

1. **Deploy a REST Function Server**: Implement a REST service that conforms to the
   `REST Function Server API specification <https://github.com/prestodb/presto/blob/master/presto-openapi/src/main/resources/rest_function_server.yaml>`_.
   The server must implement endpoints for function discovery, management, and execution.

   Key requirements:

   * Implement ``GET /v1/functions`` to list available functions
   * Implement ``POST /v1/functions/{schema}/{functionName}/{functionId}/{version}`` for function execution
   * Accept serialized input data with appropriate Content-Type:

     * ``Content-Type: application/X-presto-pages`` for Presto page format
     * ``Content-Type: application/X-spark-unsafe-row`` for Spark unsafe row format

   * Return serialized results with the same Content-Type as the request

2. **Configure the Presto C++ Worker**: Add the following to your worker's

>    configuration file (e.g., ``config.properties``):

Reviewer: ```suggestion
   configuration file (for example, ``config.properties``):
```
Avoid Latin and Latin abbreviations. See the [Gitlab documentation style guide recommended word list entry for e.g ](https://docs.gitlab.com/development/documentation/styleguide/word_list/#eg)for discussion and alternative phrasings.
